provider/clusterautoscaler/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,640 @@
+			fmt.Sprintf(
+				"Failed getting autoscaler for cluster '%s': %v",
+				state.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	populateAutoscalerState(getResponse.Body(), state.Cluster.ValueString(), state)
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterAutoscalerResource) Update(ctx context.Context, request resource.UpdateRequest,
+	response *resource.UpdateResponse) {
+	var diags diag.Diagnostics
+
+	// Get the plan:
+	plan := &ClusterAutoscalerState{}
+	diags = request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := r.collection.Cluster(plan.Cluster.ValueString()).Autoscaler().Get().SendContext(ctx)
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed getting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed getting autoscaler for cluster '%s': %v",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	autoscaler, err := clusterAutoscalerStateToObject(plan)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s: %v ",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	update, err := r.collection.Cluster(plan.Cluster.ValueString()).
+		Autoscaler().Update().Body(autoscaler).SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s': %v",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	object := update.Body()
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(object, plan.Cluster.ValueString(), state)
+
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *ClusterAutoscalerResource) Delete(ctx context.Context, request resource.DeleteRequest,
+	response *resource.DeleteResponse) {
+	state := &ClusterAutoscalerState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resource := r.collection.Cluster(state.Cluster.ValueString()).Autoscaler()
+	_, err := resource.Delete().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed deleting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed deleting autoscaler for cluster '%s': %v",
+				state.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	response.State.RemoveResource(ctx)
+}
+
+func (r *ClusterAutoscalerResource) ImportState(ctx context.Context, request resource.ImportStateRequest,
+	response *resource.ImportStateResponse) {
+	clusterId := request.ID
+
+	getResponse, err := r.collection.Cluster(clusterId).Autoscaler().Get().SendContext(ctx)
```

Comment:
why do we have a special code in import function - the import is with clusterId (only) so we can use the pasthrough

Reply:
did not understand...

Reply:
Thanks for the tip. Will update.